api/src/main/java/net/kyori/adventure/audience/Audience.java

```diff
@@ -139,9 +141,9 @@
   /**
    * Hides a bossbar.
    *
-   * @param bar the bossbar
+   * @param barId the bossbar id
    */
-  void hideBossBar(final @NonNull BossBar bar);
+  void hideBossBar(final @NonNull UUID barId);
```

Comment:
Why the change to a UUID? and why only for the `hide` operation?

We only track + control Adventure `BossBar` instances, so I don't see how it makes sense to pass an arbitrary uuid -- this feels like it will only be used as `viewer.hideBossBar(bar.id())`.

Reply:
well this allows us to hide bossbars without a reference to an instance, which would be safer. boss bars are identified by unique id as described by kashike above, so i think this is an improvement. for show operation, the contents of the bar is necessary since the bar can be new.

Reply:
Where will you get a UUID except from an instance?

Reply:
The point is to keep track of a uuid than an instance to prevent resource leaks.

Reply:
You still need to hold on to an instance to be able to change the contents of the boss bar, and boss bar listeners need to hold on to copies of the instance to ensure they remove themselves from its list of listeners. Where do you see these resource leaks coming from that would be resolved by using a UUID?

Reply:
Not all cases will make changes to a boss bar once shown, it also seems silly to enforce users to hold the entire object. Every implementation is only going to care about the `UUID`.

Reply:
An alternative idea is doing something like this:

```java
interface Id {
  UUID id();
}

interface Audience {
  hideBossBar(Id barId);
}

Audience a;
a.hideBossBar(Id.of(...));
a.hideBossBar(BossBar.of(...));
```

Reply:
As an api user I would rather not have to extract and possibly store the uuid just to remove the boss bar from the player.  I might as well just store the whole bossbar where I get the added options of modifying the content. It's making the api complicated just for a potential tiny memory saving (which I doubt will happen anyway as the implementation still has the active bossbar until all players are deregistered).

Reply:
Thought on above? ^ That way you can use either `BossBar` or `Id`

Reply:
I don't think that's worth the added complexity. Boss bars cannot be safely implemented without having the implementation hold a reference to the `BossBar` instance, and at that point a pointer is a pointer and you might as well hold on to the more useful one.

Reply:
Well you can always call the uuid version when you have the bossbar instance. like for a command node, you have a `remove(String)` while you have a `register(CommandNode)`. The fact that you don't have a `remove(CommandNode)` doesn't affect your usage at all. Same logic for uuid here.

Reply:
Still not great, what if players make their own id instance and pass it to the method, we are having to introduce additional logic whereas the type system already has this solved for us.

Reply:
> 
> 
> Still not great, what if players make their own id instance and pass it to the method, we are having to introduce additional logic whereas the type system already has this solved for us.

Equivalent scenario:
We have modified brigadier so that you can call `remove(String)` on a command node to remove children.

"Still not great, what if players make their own string instance and pass it to the method, we are having to introduce additional logic whereas the type system already has this solved for us."

Reply:
Definitely not a fan of changing these methods to accept `UUID` alone. What is the upside to allowing not a full `BossBar` to be passed?